src/crystal/system/unix/group.cr

```diff
@@ -0,0 +1,63 @@
+require "c/grp"
+
+class Crystal::System::Group
```

Comment:
I might be misunderstanding, but I don't see how `File` reopens the original type -- the base `File` class uses `include` on `Crystal::System::File`, which in turn references some constants from `::File` but doesn't reopen it. 

(I agree with the overarching point about unnecessary duplication, I just want to make sure I understand your recommendation :smile:)

Reply:
Oh sry, `File` was an example for including a `Crystal::System` type. That might be a good approach here, too. But not two separate types with delegation.

Reply:
Alright, refactored so that `User` and `Group` each `extend` their `Crystal::System` counterpart (`include` doesn't work because we're mixing class methods in).

One slight problem: `from_name?` and `from_id?` don't appear on their extending classes in the documentation. The docs correctly show that e.g. `Crystal::System::User` is being extended, but don't provide documentation for it (correctly, IMO).

Is there a workaround to this? I suppose I could just move the `?` methods out of `Crystal::System` and into `System` and have them call helper methods, but that feels a little clunky.